static/js/components/admin/EditChannelMembersForm.js

```diff
@@ -3,32 +3,113 @@ import React from "react"
 import { Link } from "react-router-dom"
 
 import { MISSING_TEXT } from "../../lib/channels"
+import { validationMessage } from "../../lib/validation"
 import { profileURL } from "../../lib/url"
 
-import type { Member } from "../../flow/discussionTypes"
+import type {
+  AddMemberForm,
+  AddMemberValidation,
+  Channel,
+  Member
+} from "../../flow/discussionTypes"
 
 type Props = {
   members: Array<Member>,
-  usernameGetter: (member: Member) => string
+  usernameGetter: (member: Member) => string,
+  addMember: Function,
+  removeMember: Function,
+  updateEmail: Function,
+  channel: Channel,
+  validation: AddMemberValidation,
+  form: AddMemberForm,
+  processing: boolean,
+  beginFormEdit: Function,
+  noun: string
```

Comment:
Could you give this a more descriptive variable name? It doesn't tell me a lot about what it is and since it's being passed in via state through `withMemberForm`, it can be unclear at the page container level what a state property `noun` is supposed to be for.